starknet-rpc-test/get_transaction_by_blockid_and_index.rs

```diff
@@ -0,0 +1,140 @@
+    let rpc = madara.get_starknet_client();
+
+    assert_matches!(
+            rpc
+            .get_transaction_by_block_id_and_index(
+                BlockId::Tag(BlockTag::Latest),
+                0
+            )
+            .await,
+            Err(StarknetProviderError(StarknetErrorWithMessage { code:
+    MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::InvalidTransactionIndex
+        );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn work_ok_by_compare_with_get_block_with_tx(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    let signer = LocalWallet::from(SigningKey::from_secret_scalar(FieldElement::from_hex_be(SIGNER_PRIVATE).unwrap()));
+    let argent_account_address = FieldElement::from_hex_be(ARGENT_CONTRACT_ADDRESS).expect("Invalid Contract Address");
+    let account = SingleOwnerAccount::new(rpc, signer, argent_account_address, chain_id::TESTNET);
+
+    madara.create_empty_block().await?;
+
+    let execution_1 = account.transfer_tokens(
+        argent_account_address,
+        FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+        None,
+    );
+
+    let execution_2 = account
+        .transfer_tokens(
+            FieldElement::from_hex_be(TEST_CONTRACT_CLASS_HASH).expect("Invalid Contract Address"),
+            FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+            None,
+        )
+        .nonce(FieldElement::ONE)
+        .max_fee(FieldElement::from_hex_be("0xDEADB").unwrap());
+
+    madara
+        .create_block_with_txs(vec![TransactionEnum::Execution(execution_1), TransactionEnum::Execution(execution_2)])
+        .await?;
+
+    let tx_1 = rpc.get_transaction_by_block_id_and_index(BlockId::Tag(BlockTag::Latest), 0).await?;
+    let tx_2 = rpc.get_transaction_by_block_id_and_index(BlockId::Tag(BlockTag::Latest), 1).await?;
+
+    // Known tx1 and tx2 are InvokeTransactionV1
+    assert!(matches!(tx_1, Transaction::Invoke(InvokeTransaction::V1(_))));
+    assert!(matches!(tx_2, Transaction::Invoke(InvokeTransaction::V1(_))));
+
+    let tx_1_obj = get_transaction_obj(&tx_1).expect("Invalid Transaction");
+    let tx_2_obj = get_transaction_obj(&tx_2).expect("Invalid Transaction");
+
+    assert_eq!(tx_1_obj.nonce, FieldElement::ZERO);
+    assert_eq!(tx_1_obj.sender_address, argent_account_address);
+
+    assert_eq!(tx_2_obj.nonce, FieldElement::ONE);
+    assert_eq!(tx_2_obj.sender_address, argent_account_address);
+    assert_eq!(tx_2_obj.max_fee, FieldElement::from_hex_be("0xDEADB").unwrap());
+
+    let block_with_txs = rpc.get_block_with_txs(BlockId::Tag(BlockTag::Latest)).await?;
+
+    assert_eq!(Some(&tx_1_obj.transaction_hash), get_transaction_hash_from_block_with_txs(&block_with_txs, 0));
+    assert_eq!(Some(&tx_2_obj.transaction_hash), get_transaction_hash_from_block_with_txs(&block_with_txs, 1));
+
```

Comment:
You can probably use assert matches here too. With a really big nested match arm, but it will spare you all those helper functions